pyproject.toml

```diff
@@ -7,8 +7,8 @@ authors = ["Dan Bentley <dan@numiko.com>"]
 [tool.poetry.dependencies]
 python = "^3.8"
 gunicorn = "^20.1.0"
-Django = "~4.0"
-wagtail = "~3.0"
+Django = "^4.1"
+wagtail = "^4.0.2"
```

Comment:
Why choose `^` over `~` ?

Reply:
When ^ is used then an update is allowed if the new version number does not modify the left-most non-zero digit in the major, minor, patch grouping. The left-most non-zero is 4 here
So ^4.1 means >=4.1.0 <5.0.0

Reply:
Thanks for the response.

https://docs.djangoproject.com/en/dev/internals/release-process/#django-s-release-process
> A.B is the feature release version number. Each version will be mostly backwards compatible with the previous release. Exceptions to this rule will be listed in the release notes 

Minor release is not always 100% compatible and may require testing or even code changes in this case. It could be safer to use `~` .

>  ~1.2.3 this means >=1.2.3 < 1.3.0 , here we have major, minor and patch, only patch is allowed

This is what the existing/earlier version of `pyproject.toml` code base is doing.